src/main/java/com/arpnetworking/metrics/common/parsers/Parser.java

```diff
@@ -38,5 +38,5 @@
      * @return Instance of <code>Record</code> from the data.
      * @throws ParsingException If parsing of the data fails for any reason.
      */
-    T parse(byte[] data) throws ParsingException;
+    T parse(final byte[] data) throws ParsingException;
```

Comment:
Does this pass checkstyle? Final use in interfaces is supposed to be a negative since it has no implication on the contract.